massa-execution-worker/src/execution.rs

```diff
@@ -1507,17 +1508,37 @@ impl ExecutionState {
 
         // return the execution output
         let execution_output = context_guard!(self).settle_slot(None);
-        let exact_cost = req.max_gas.saturating_sub(exec_response.remaining_gas);
+        let gas_used_until_last_subexecution = (context_guard!(self))
+            .gas_remaining_before_subexecution
+            .unwrap_or_default(); // 0 if no subexecutions
+        let exact_exec_cost = req.max_gas.saturating_sub(exec_response.remaining_gas);
+        // compute a gas cost, estimating the gas of the last SC call to be max_instance_cost
+        let corrected_cost = req
+            .max_gas
+            .saturating_sub(gas_used_until_last_subexecution)
+            .saturating_add(self.config.gas_costs.max_instance_cost);
+        // keep the max of the two so the last SC call has at least max_instance_cost of gas
```

Comment:
I also have a doubt about "launch_cost", you might need to take a look whether it should or not be counted here

Reply:
clarified